src/transform_output/transform_output.cpp

```diff
@@ -68,149 +68,199 @@ void transform_output(const char *ip_file_name, const char *op_file_name) {
+					 * e.g : token_1 = [iestream_input_defs<message_t>::out: {5}
+					 *       token_2 = generated by model generator_con
+					 */
+					char* token = strtok(line, "]"); 
+	  				char* token_1 = token,* token_2 =strtok(NULL, "]");
+	  				
+	  				/**
+	  				 * divide token_2 by delimiter space (" ") to extract component part from it.
+	  				 * token_6 holds component part at the end of while loop.
+	  				 */
+	  				char* token_5 = strtok(token_2, " "), * token_6;
+	  				
+	  				/**
+				     * Keep printing tokens while one of the 
+				     * delimiters present in token_2. 
+				     */
+				    while (token_5 != NULL) { 
+				        component = token_5;
+				        token_5 = strtok(NULL, " ");
+				    } 
+				    
+				    /**
+				     * divide token_1 by delimiter comma (",") to extract all port and its corresponding values
+				     * while loop runs until all ports and its values are obtained
+				     * for each token retrieved in while loop, we check it with port constant to check if a port is
+				     * present in the token. If yes then we assign the port to string variable "port"
+				     */
+				    char* token_3 = strtok(token_1, ",");
+				    char* token_4;
 
-					}else if(strstr(token_3,"packetSentOut")){
+				    /**
+				     * Keep printing tokens while one of the 
+				     * delimiters present in token_3.
+				     */ 
+				    while (token_3 != NULL) { 
 
-					 	port="packetSentOut";
+				    	if(strstr(token_3,"out")){
 
-					} else if(strstr(token_3,"ackReceivedOut")){
+							port= "out";
+							
+						}else if(strstr(token_3,"dataOut")){
 
-					 	port="ackReceivedOut";
-					}
-					
-					/**
-					 * for each token_3, we extract the port value by using strstr() with "{" delimiter.
-					 * port value is present in a "{}" brackets.
-					 */
-					
-					token_4 = strstr(token_3,"{");
+							port="dataOut";
+
+						}else if(strstr(token_3,"packetSentOut")){
+
+						 	port="packetSentOut";
+
+						} else if(strstr(token_3,"ackReceivedOut")){
+
+						 	port="ackReceivedOut";
+						}
+						else if (component== "generator_con") {
+							component_gen_c = "generator_con";
+						}
+						
+						/**
+						 * for each token_3, we extract the port value by using strstr() with "{" delimiter.
+						 * port value is present in a "{}" brackets.
+						 */
+						token_4 = strstr(token_3,"{");
+						
+						/**
+					 	 * if ports value is empty then we ignore such ports and 
+						 * continue. This is verifed if the length of token_4 is greater than 2.
+						 * if a value is obtained other than empty then we assign it to port_value variable.
+						 */
+			            if (strlen(token_4)>2){
+							
+							in = token_4;
+							port_value = in.substr( 1 , ( in.find_first_of('}', 0) - 1) );
+			        			/** 
+								* Check Alternating Bit status,only if it is equal to 1
+								*/
+								if (port_value.length() ==1 ){

```

Comment:
I think there should be 4 spaces removed

Reply:
Modified as requested